### PR TITLE
Change the default database to PickledDB

### LINKED ISSUE #572: Set PickledDB as default Storage Backend
Note: `docs/src/install/database.rst` should be reworked based on new default backend. Running a grep for `mongodb` in docs would be a good idea also to make sure we don't miss any other docs that need an update.

To be clear, this task involves mostly work on the documentation. In the code there is probably only `src/orion/core/__init__.py` that needs to be modified.

### PR DESCRIPTION
**Pull request by abergeron**
# Description
Fixes #572 

# Checklist
## Tests
- [ ] All new and existing tests are passing (`$ tox -e py38`; replace `38` by your Python version if necessary)

## Documentation
- [x] I have updated the relevant documentation related to my changes

## Quality
- [x] I have read the [CONTRIBUTING](https://github.com/Epistimio/orion/blob/develop/CONTRIBUTING.md) doc
- [x] My commits messages follow [this format](https://chris.beams.io/posts/git-commit/)
- [x] My code follows the style guidelines (`$ tox -e lint`)

### CODE REVIEW

**Comment by abergeron** [comment]
It seems that the MongoDB setup failed, but I didn't change anything there.  I can't manually re-run the tests.

Should I push an empty commit to restart the tests?

**Comment by abergeron** [comment]
This doesn't pass the tests currently apparently, but I would probably wait for #582 before I fix it.

File: docs/src/install/database.rst
Comment (bouthilx) [inline docs/src/install/database.rst:79]: You can drop `'name'` for the pickleddb.

File: docs/src/install/database.rst
Comment (bouthilx) [inline docs/src/install/database.rst:98]: Here the text should be adjusted. Perhaps make it an example using mongodb, because there isn't much options to set on pickleddb.

File: docs/src/user/config.rst
Comment (bouthilx) [inline docs/src/user/config.rst:92]: name and port can be dropped.

File: docs/src/user/config.rst
Comment (bouthilx) [inline docs/src/user/config.rst:177]: The default is https://github.com/Epistimio/orion/blob/develop/src/orion/core/io/database/pickleddb.py#L25

File: src/orion/core/__init__.py
Comment (bouthilx) [inline src/orion/core/__init__.py:115]: Default should be https://github.com/Epistimio/orion/blob/develop/src/orion/core/io/database/pickleddb.py#L25

File: docs/src/install/database.rst
Comment (abergeron) [reply docs/src/install/database.rst:79]: Actually I would add 'port' since it would match what orion outputs.

I agree that they are irrelevant for pickleddb, but it might confuse new users to see unexpected output.

File: docs/src/user/config.rst
Comment (abergeron) [reply docs/src/user/config.rst:177]: Maybe but trying to import that file in the config function leads to an import loop and fails.

File: src/orion/core/__init__.py
Comment (abergeron) [reply src/orion/core/__init__.py:115]: Maybe but trying to import that file in the config function leads to an import loop and fails.

Comment (abergeron) [reply src/orion/core/__init__.py:115]: But I found out that I can just reach into DIRS locally and use that to build the default path so that works.

**Comment by abergeron** [comment]
I've made the changes locally, but I'm waiting on the merge of #582 to push them since I've rebased this branch on top of that.

File: src/orion/core/__init__.py
Comment (bouthilx) [reply src/orion/core/__init__.py:115]: Actually you can forget about this default. I prefer Olivier's idea to set it to None by default and leave it to the DB class to set it to DEFAULT if it is None. Could you please do this in this PR?

Comment (abergeron) [reply src/orion/core/__init__.py:115]: Ok

Comment (abergeron) [reply src/orion/core/__init__.py:115]: It actually doesn't work and sets the default db to the string "None" if I put `None` there.

Comment (bouthilx) [reply src/orion/core/__init__.py:115]: Yes that's what I meant by 'leave it to the DB class to set it to DEFAULT if it is None. Could you please do this in this PR?'

You would need to adjust the code here: https://github.com/Epistimio/orion/blob/develop/src/orion/core/io/database/pickleddb.py#L100

Comment (abergeron) [reply src/orion/core/__init__.py:115]: The 'None' string was due to Configuration, but I've adjusted to make the default an empty string and the database classes will replace that with their built-in defaults.

**Comment by bouthilx** [comment]
@abergeron The PR was merged. This is ready for rebase.

**Review by bouthilx — APPROVED** [review]

**Pull request merged**
